src/diffusers/models/cross_attention.py

```diff
@@ -71,23 +77,47 @@ def __init__(
         self.sliceable_head_dim = heads
 
         self.added_kv_proj_dim = added_kv_proj_dim
-
+        
         if norm_num_groups is not None:
             self.group_norm = nn.GroupNorm(num_channels=inner_dim, num_groups=norm_num_groups, eps=1e-5, affine=True)
         else:
             self.group_norm = None
-
-        self.to_q = nn.Linear(query_dim, inner_dim, bias=bias)
-        self.to_k = nn.Linear(cross_attention_dim, inner_dim, bias=bias)
-        self.to_v = nn.Linear(cross_attention_dim, inner_dim, bias=bias)
-
+        
+        if cross_attention_norm:
+            self.norm_cross = nn.LayerNorm(cross_attention_dim)
+        
+        if use_conv_proj:
+            self.to_q = nn.Conv2d(query_dim, inner_dim, 1)
```

Comment:
`to_q` should always be linear (there are identical for attention -> we can change this later though :-)